ordering/__init__.py

```diff
@@ -68,11 +68,13 @@ def __contains__(self, item: _T) -> bool:
         return item in self._labels
 
     def __iter__(self) -> Iterator[T]:
-        item = self._successors[self._start]
-
-        while not isinstance(item, _Sentinel):
+        successors = self._successors
+        item = successors[self._start]
+        end = self._end
+        while item is not end:
+            assert not isinstance(item, _Sentinel)
             yield item
-            item = self._successors[item]
+            item = successors[item]
```

Comment:
IIRC, the variable was originally named something like `nexts` (to echo the `next` built-in), I renamed it for consistency with the other times I'd done the same in more complex functions like `swap` (or my experiments with slicing) where extracting the internal dicts to local variables did definitely help with readability.

It's also probably a product of my mind associating `_successors`, etc. more strongly with the item than the Ordering itself, this would be another of my driving reasons for wanting to rewrite Ordering in the style of OrderedDict, as the internal dicts would be morphed into attributes of OrderingItem. 

In that event things in general would become considerably more readable, as in `self._successors[self._predecessors[existing_item]]` would become the much easier to parse `existing_item.prev.next`. Or more relevantly `__iter__` would look like:

```Python
def __iter__(self) -> Iterator[T]:
    stop = self.__end
    item = self.__start.next
    
    while item is not stop:
        yield item.item
        item = item.next
```